client/packages/invoices/src/OutboundShipment/DetailView/OutboundLineEdit/allocation/allocateQuantities.ts

```diff
@@ -71,130 +75,152 @@ export const allocateQuantities = (
+    allocateIn,
   });
 
   // Step 2: if still some remaining quantity allocate to the nearest (rounded up) pack size
-  if (toAllocate > 0) {
-    toAllocate = allocateToBatches({
+  if (quantityToAllocate > 0) {
+    quantityToAllocate = allocateToBatches({
       validBatches,
       newDraftLines,
-      toAllocate,
+      quantityToAllocate,
+      allocateIn,
       roundUp: true,
     });
   }
 
-  // Step 3: if over-allocated (negative toAllocate), reduce the smaller/earlier
+  // Step 3: if over-allocated (negative quantityToAllocate), reduce the smaller/earlier
   // lines to try to get back to the requested quantity
-  if (toAllocate < 0) {
-    toAllocate = reduceBatchAllocation({
-      toAllocate: toAllocate * -1,
+  if (quantityToAllocate < 0) {
+    quantityToAllocate = reduceBatchAllocation({
+      quantityToAllocate: quantityToAllocate * -1,
       validBatches,
       newDraftLines,
+      allocateIn,
     });
   }
 
   return {
     allocatedLines: newDraftLines,
-    remainingQuantity: toAllocate,
+    remainingQuantity: quantityToAllocate,
   };
 };
 
 const allocateToBatches = ({
   validBatches,
   newDraftLines,
-  toAllocate,
+  quantityToAllocate: remainingQuantityToAllocate,
+  allocateIn,
   roundUp = false,
 }: {
   validBatches: DraftStockOutLineFragment[];
   newDraftLines: DraftStockOutLineFragment[];
-  toAllocate: number;
+  quantityToAllocate: number;
+  allocateIn: AllocateIn;
   roundUp?: boolean;
 }) => {
   validBatches.forEach(batch => {
-    if (toAllocate <= 0) return null;
+    if (remainingQuantityToAllocate <= 0) return null;
 
-    const draftOutboundLineFragmentIdx = newDraftLines.findIndex(
-      ({ id }) => batch.id === id
-    );
-    const draftOutboundLineFragment =
-      newDraftLines[draftOutboundLineFragmentIdx];
+    const draftLineIndex = newDraftLines.findIndex(({ id }) => batch.id === id);
+    const draftLine = newDraftLines[draftLineIndex];
 
-    if (!draftOutboundLineFragment) return null;
+    if (!draftLine) return null;
 
-    const {
-      availablePacks,
-      packSize,
-      numberOfPacks: numPacksAlreadyAllocated,
-    } = draftOutboundLineFragment;
+    // helpers
+    const toQuantity = (packs: number) =>
+      packsToQuantity(allocateIn, packs, draftLine);
+    const toPacks = (quantity: number) =>
+      quantityToPacks(allocateIn, quantity, draftLine);
 
-    const availableUnits =
-      Math.floor(availablePacks - numPacksAlreadyAllocated) * packSize;
+    // TODO: Allow partial packs check would be needed before .floor() here
+    const allocatablePacks = Math.floor(
+      // remove already allocated packs from available
+      draftLine.availablePacks - draftLine.numberOfPacks
+    );
 
-    const unitsToAllocate = Math.min(toAllocate, availableUnits);
-    const numberOfPacksToAllocate = unitsToAllocate / packSize;
+    const quantityToAllocate = Math.min(
+      remainingQuantityToAllocate,
+      toQuantity(allocatablePacks)
+    );
```

Comment:
Sorry for diff being a bit noisy in here, tried to clean up naming a bit. Logical change is just the to and from `quantity` (either units or doses based on allocateIn)